### DISCUSSION TITLE
Colour for ScrollBar in ScrollView

### DISCUSSION THREAD

**Discussion by ThanhHaiDang96** [dotnet/maui#20992] [Ideas]
### Description

Hi,

How can I set color for scroll bar in a ScrollView?
I have a VerticalStackLayout inside a ScrollView. The StackLayout has background is Black, so I cannot show the ScrollBar with default color. 

Do you have any recommends for this issue?

### Public API Changes

``` CSharp
public partial class ScrollView 
{
// ...
public static readonly BindableProperty ScrollBarColorProperty = BindableProperty.Create(nameof(ScrollBarColor), typeof(Color), typeof(ScrollView), default(Color));

public Color ScrollBarColor
{
get {return (Color) GetValue(ScrollBarColorProperty );}
set {SetValue(ScrollBarColorProperty , value};
}
//...
}
```

### Intended Use-Case

In my app, I have a VerticalStackLayout inside a ScrollView. The StackLayout has background is Black, so I cannot show the ScrollBar with default color.

**Comment by davidortinau** [comment]
You can access the platform properties and views to account for this. For example on iOS / Mac Catalyst:

https://developer.apple.com/documentation/uikit/uiscrollviewindicatorstyle?language=objc

```csharp
using Microsoft.Extensions.Logging;
using Microsoft.Maui.Platform;

#if IOS || MACCATALYST
using UIKit;
#endif

namespace ScrollBarColor;

public static class MauiProgram
{
	public static MauiApp CreateMauiApp()
	{
		var builder = MauiApp.CreateBuilder();
		builder
			.UseMauiApp<App>()
			.ConfigureMauiHandlers(handlers =>
			{
				ThemeScrollBar();
			})
			.ConfigureFonts(fonts =>
			{
				fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
				fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
			});

#if DEBUG
		builder.Logging.AddDebug();
#endif

		return builder.Build();
	}

	private static void ThemeScrollBar()
	{
		Microsoft.Maui.Handlers.ScrollViewHandler.Mapper.AppendToMapping("MyCustomization", (handler, view) =>
        {
			
#if IOS
			handler.PlatformView.Scrolled += (s, e) =>
			{
				var scrollView = (UIScrollView)s;
				var verticalIndicator = scrollView.Subviews.Last();
    			verticalIndicator.BackgroundColor = UIColor.Red;
				
			};
#elif MACCATALYST
			handler.PlatformView.IndicatorStyle = UIScrollViewIndicatorStyle.White;
#elif ANDROID
			// Android customization
#elif WINDOWS	
			// Windows customization
#endif
		});
	}
}
```

https://github.com/dotnet/maui/assets/41873/386f6ac2-e466-4b7e-a93d-a343cbf4815c


https://github.com/dotnet/maui/assets/41873/db5da36a-b79e-4da8-9a6e-d988b66a4924

**Reply by ThanhHaiDang96** [reply]
Hi [davidortinau](https://github.com/davidortinau),

Thank for your suggestion. However, I just want to change the color in some special cases, and keep default in the others.

Regards,
TH

**Reply by HAdennishellen** [reply]
Hi @davidortinau,

You had solutions for IOS and Mac, but do you have a solution for android?

Regards Dennis H

**Reply by amortensen13** [reply]
Thanks for the code.  This helped some.  The scrollbar is now gray, so it shows in dark mode and light mode.  Would prefer to be able to customize it for every ScrollView, since some screens in my app are dark and some are light, but at least now I can see the scroll bar (while I'm scrolling) on both screens.

**Reply by granevich** [reply]
Does anyone have the solution for Windows?
